pydantic/validators.py

```diff
@@ -217,9 +217,10 @@ def set_validator(v: Any) -> Set[Any]:
 
 
 def enum_validator(v: Any, field: 'Field', config: 'BaseConfig') -> Enum:
-    with change_exception(errors.EnumError, ValueError):
```

Comment:
Why don't just add kwargs support to context manager?
and then:
```py
with change_exception(errors.EnumError, ValueError, enum_type=field.type_):
    ...
```

Reply:
I'm guessing that would be better placed in a separate PR; not sure it would be worth it if it was only used in this instance. But I haven't looked to see if it could simplify the error handling anywhere else.